sdcm/utils/database_query_utils.py

```diff
@@ -150,15 +151,25 @@ def validate_rows_per_partitions(self, ignore_limit_rows_number: bool = False):
             LOGGER.debug('Validate partitions info')
             partitions_dict_after = self.collect_partitions_info(ignore_limit_rows_number=ignore_limit_rows_number)
             if partitions_dict_after is not None:
-                if not ignore_limit_rows_number and self.limit_rows_number:
-                    missing_rows = {key: val for key, val in partitions_dict_after.items() if
-                                    val < self.limit_rows_number}
-                    assert not missing_rows, f"Found missing rows for partitions: {missing_rows}"
-                else:
-                    self.tester.assertEqual(self.partitions_dict_before,
-                                            partitions_dict_after,
-                                            msg='Row amount in partitions is not same before and after running of nemesis: '
-                                                f' {partitions_dict_after}')
+                try:
+                    if not ignore_limit_rows_number and self.limit_rows_number:
+                        missing_rows = {key: val for key, val in partitions_dict_after.items() if
+                                        val < self.limit_rows_number}
+                        assert not missing_rows, f"Found missing rows for partitions: {missing_rows}"
+                    else:
+                        self.tester.assertEqual(self.partitions_dict_before,
+                                                partitions_dict_after,
+                                                msg='Row amount in partitions is not same before and after running of nemesis: '
+                                                    f' {partitions_dict_after}')
+                except AssertionError as error:
+                    PartitionRowsValidationEvent(
+                        message=str(error),
+                        severity=Severity.CRITICAL).publish()
```

Comment:
why critical event ? is should stop the test as the first time this is happening ?

Reply:
i'm not aware of something more critical for customer than data validation issue.
SCT defines c-s load failure as critical, which is less problematic than this error.

Reply:
Keep in mind that every time it stops a test, you will be called to help decipher what going on